tests/test_utils.py

```diff
@@ -169,3 +174,29 @@ def check_standard_result_data(i, job, result):
     assert result.result_type == ResultType.get_result_type(job.type)
     assert result.job_id == job.id
     assert result.circuit == job.deployment.programs[i].quantum_circuit
+
+
+def check_if_job_runner_result_correct_multiple_gates(job: JobDataclass):
+    """Iterate over every result and check if the distribution of the measurement is correct"""
+
+    for i in range(len(job.results)):
+        result: ResultDataclass = job.results[i]
+        check_standard_result_data(i, job, result)
+        assert result.meta_data is not None
+        shots: int = job.shots
+        counts: dict = result.result_dict["counts"]
+        probabilities: dict = result.result_dict["probabilities"]
+        prob_tolerance: float = PROBABILITY_TOLERANCE * 2
+        count_tolerance: float = COUNTS_TOLERANCE * 2
+        if i != 2:
+            qubit = QUBIT_8 if i == 3 else QUBIT_1
+            assert counts[qubit] == shots
+            assert probabilities[qubit] == PROBABILITY_1
+        else:
+            assert compare_values_with_tolerance(7 * (shots / 8), counts[QUBIT_0], count_tolerance)
```

Comment:
maybe relocate (auslagern) the assertions in else under the call "compare_values" ot sth like that (function call and for loop are huge)